spark-java/src/main/java/com/baeldung/sparkjava/UserServiceMapImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.baeldung.sparkjava;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+
+public class UserServiceMapImpl  implements UserService{
+    private HashMap<String, User> userMap;
+    
+    public UserServiceMapImpl() {
+        userMap = new HashMap<>();
+    }
+
+    public void addUser (User user) {
+        userMap.put(user.getId(), user);
+    }
+    
+    public Collection<User> getUsers () {
+        return  userMap.values();
+    }
+    
+    public User getUser (String id) {
+        return userMap.get(id);
+    }
+    
+    public User editUser (String id, HashMap userArg) throws Exception{
+        User toEdit = userMap.get(id);
+        if (toEdit == null )
+            return null;
+        toEdit.setEmail((userArg.get("email")!=null) ? (String) userArg.get("email") : toEdit.getEmail() );
```

Comment:
Done. I thought it would not look "cool" to have so many if's in the function. But then I realized as you rightly guided, we cannot compromise simplicity of the code in making it look more fancy. Of late I realized, rather than fancy, it looked more complicated. Thanks for guiding.